WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -944,124 +939,93 @@ class AbstractPostListViewController: UIViewController,
         navigationController?.present(navWrapper, animated: true)
     }
 
-    @objc func deletePost(_ apost: AbstractPost) {
-        WPAnalytics.track(.postListTrashAction, withProperties: propertiesForAnalytics())
-
-        let postObjectID = apost.objectID
+    @MainActor
+    func deletePost(_ apost: AbstractPost) async {
+        assert(apost.managedObjectContext == ContextManager.shared.mainContext)
```

Comment:
**Out of scope**: I've been thinking about all the accesses to `ContextManager.shared` that we have and the issues they sometime cause such as https://github.com/wordpress-mobile/WordPress-iOS/issues/21742.

Would it make sense to instantiate the view controller with a context or `CoreDataStack` object? In practice, this would be `ContextManager.shared.mainContext` or `ContextManager.shared` but the separation would:

1. Centralize the definition of the shared resource access
2. Allow us to replace it in the tests

What do you think?